pkg/deviceplugin/server.go

```diff
@@ -163,15 +174,15 @@ func (srv *server) Update(devices map[string]DeviceInfo) {
 func (srv *server) setupAndServe(namespace string, devicePluginPath string, kubeletSocket string) error {
 	resourceName := namespace + "/" + srv.devType
 	pluginPrefix := namespace + "-" + srv.devType
+	srv.state = serving
 
-	for {
+	for srv.state == serving {
 		pluginEndpoint := pluginPrefix + ".sock"
 		pluginSocket := path.Join(devicePluginPath, pluginEndpoint)
 
 		if err := waitForServer(pluginSocket, time.Second); err == nil {
 			return errors.Errorf("Socket %s is already in use", pluginSocket)
 		}
-		os.Remove(pluginSocket)
```

Comment:
This line should be better left as it is. If I remember correctly it cleans up dead sockets left after crashes.